subprojects/model-core/src/main/java/org/gradle/internal/properties/bean/DefaultPropertyWalker.java

```diff
@@ -85,8 +84,8 @@ public void visitNestedUnpackingError(String qualifiedName, Exception e) {
             }
 
             @Override
-            public void visitLeaf(String qualifiedName, PropertyMetadata propertyMetadata, Supplier<Object> value) {
-                PropertyValue cachedValue = new CachedPropertyValue(value, propertyMetadata.getGetterMethod());
+            public void visitLeaf(Object parent, String qualifiedName, PropertyMetadata propertyMetadata) {
+                PropertyValue cachedValue = new CachedPropertyValue(() -> propertyMetadata.getPropertyValue(parent), propertyMetadata.getDeclaredType().getRawType());
```

Comment:
❓ Why we are passing the parent node around? It seems more work for a caller, than just getting `supplier.get()` or something similar.

Reply:
From Slack: This will change with next PR.